lms/djangoapps/discussion_api/api.py

```diff
@@ -108,15 +108,53 @@ def _is_user_author_or_privileged(cc_content, context):
     )
 
 
-def get_thread_list_url(request, course_key, topic_id_list):
+def get_thread_list_url(request, course_key, topic_id_list=None):
     """
     Returns the URL for the thread_list_url field, given a list of topic_ids
     """
     path = reverse("thread-list")
-    query_list = [("course_id", unicode(course_key))] + [("topic_id", topic_id) for topic_id in topic_id_list]
+    query_list = (
+        [("course_id", unicode(course_key))] +
+        [("topic_id", topic_id) for topic_id in topic_id_list or []]
+    )
```

Comment:
We will run into some browsers' limits once we hit about 40 leaves under a single intermediate node (~2k characters at 32 characters per topic_id plus parameter names and other overhead). I don't know if django has a URL length limit. And yes, this is an old problem.